osconfig_tests/config/config.go

```diff
@@ -0,0 +1,52 @@
+package config
+
+import (
+	"fmt"
+	"time"
+)
+
+var (
+	// TODO: allow this to be configurable through flag to test against staging
+	prodEndpoint    = "osconfig.googleapis.com:443"
+	oauthDefault    = ""
+	bucketDefault   = "osconfig-agent-end2end-tests"
+	logsPathDefault = "logs"
+
+	TestConfig *config
+)
+
+type config struct {
+	svcEndpoint, oauthPath, logsBucket, logsPath string
+}
+
+// SetConfig sets the test configs
+func SetConfig() error {
```

Comment:
i had the idea of adding an interface to provide custom location for agent downloads. but i think you are right, that can also be a flag. i do not see any use case where we need to be passed to test runner through any value that is not static.

will change it.

Reply:
ok.. i am adding this back. I am using "logspath" which has the time at which the tests are run. using config struct to store the logspath. 
i dont think there is a harm in keeping the config

Reply:
Just make that a var as well, the issue is your running this initialization right at main, so really there's no point in the struct since its only setup once at start, and it introduces the ability for you to make a nil pointer dereference.

Reply:
i am removing the struct. but to address the problem of nil pointer dereference, i think DI is the correct way to address it.